### PR TITLE
improve error message

### PR DESCRIPTION
**Pull request by superyyrrzz**
1. can know which project meets error in extracting metadata.
2. can know which file contains invalid cross reference.

### CODE REVIEW

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadataWorker.cs [hunk 1]
```diff
@@ -445,8 +445,15 @@ private static IEnumerable<string> GetTransitiveProjectReferences(ConcurrentDict
...
-                var compilation = await project.Value.GetCompilationAsync();
-                compilations.GetOrAdd(project.Key, compilation);
+                try
+                {
+                    var compilation = await project.Value.GetCompilationAsync();
+                    compilations.GetOrAdd(project.Key, compilation);
+                }
+                catch (Exception e)
+                {
+                    throw new ExtractMetadataException($"Error extracting metadata for project \"{project.Key}\": {e}", e);
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadataWorker.cs]: >e [](start = 115, length = 1)

e.Message?

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs [hunk 2]
```diff
@@ -159,7 +159,7 @@ internal ManifestItem Transform(InternalManifestItem item)
             if (missingUids.Count > 0)
             {
                 var uids = string.Join(", ", missingUids.Select(s => $"\"{s}\""));
-                Logger.LogWarning($"Unable to resolve cross-reference {uids}");
+                Logger.LogWarning($"Unable to resolve cross-reference {uids}. Referenced by file: {item.LocalPathFromRoot}");
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs]: >$"Unable to resolve cross-reference {uids}. Referenced by file: {item.LocalPathFromRoot}" [](start = 34, length = 89)

@herohua seems we better add the file info into logitem.File property

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs (hunk 2 above)
Comment (herohua) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs]: seems @superyyrrzz has fixed this

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**